src/snapred/backend/dao/state/CalibrantSample/CalibrantSamples.py

```diff
@@ -16,7 +16,7 @@ class CalibrantSamples(BaseModel):
     date: Optional[str] = None
     geometry: Geometry
     material: Material
-    crystallography: Crystallography
+    crystallography: Optional[Crystallography]
```

Comment:
why is this optional?  I'd like to avoid optional if possible.

Reply:
These are supposed to integrate easily with mantid.  Inside mantid, the sample is set using the appropriately named [`SetSample` algorithm](https://docs.mantidproject.org/nightly/algorithms/SetSample-v1.html).  However, this is only able to set the geometry and material properties of the sample (plus its environment and container).  It can't set the crystal structure of the container.  I think [`SetUB`](https://docs.mantidproject.org/nightly/algorithms/SetUB-v1.html) is the only algorithm that can let a user set the crystal structure, but so far we haven't called it.  I think it simplifies data handling having the crystallography here, but it's also only occasionally (and so far never) needed.

Reply:
It looks like the [`CrystalStructure`](https://docs.mantidproject.org/nightly/concepts/CrystalStructureAndReflections.html) data object, mixed with `ReflectionGenerator` object would also be able to use these, but so far we haven't worked with those, either, and their use seems pretty distinct from setting the calibration sample.

Reply:
We should not use `SetUB` for calibrant samples as that's specifically for single-crystal materials. I think CrystalStruture is the way to go for our powder samples.